app/components/submission/WithCurrentSubmission.js

```diff
@@ -89,42 +89,47 @@ const UPDATE_SUBMISSION = gql`
 `
 
 class WithCurrentSubmission extends React.Component {
-  constructor() {
-    super()
-    this.state = { error: null, data: null, loading: true }
-  }
+  state = { error: null, data: null, loading: true }
+
   componentDidMount() {
-    this.props.client
-      .query({ query: GET_CURRENT_SUBMISSION })
-      .then(({ data }) => {
-        if (data.currentSubmission) {
-          return this.setState({ data, error: null, loading: false })
-        }
-        return this.props.client
-          .mutate({
-            mutation: CREATE_SUBMISSION,
-            refetchQueries: [
-              {
-                query: GET_CURRENT_SUBMISSION,
-              },
-            ],
-          })
-          .then(({ data: createData }) => {
-            this.setState({
-              data: createData,
-              error: undefined,
-              loading: false,
-            })
-          })
-      })
-      .catch(error => this.setState({ error, data: null, loading: false }))
+    this.querySubscription = this.props.client
+      .watchQuery({ query: GET_CURRENT_SUBMISSION })
+      .subscribe(
+        ({ data }) => {
+          if (data.currentSubmission) {
+            this.setData(data)
+          } else {
+            this.props.client
+              .mutate({
+                mutation: CREATE_SUBMISSION,
+                refetchQueries: [{ query: GET_CURRENT_SUBMISSION }],
+              })
+              .then(result => this.setData(result.data))
+              .catch(error => this.setError(error))
+          }
+        },
+        error => this.setError(error),
+      )
+  }
+
+  setData(data) {
```

Comment:
Do we have something to deal with this? According to [transform-class-properties](https://babeljs.io/docs/en/babel-plugin-transform-class-properties/) you should have something like this

```
setData = (data) => {...}
```

Same goes for `setError()`.

Reply:
You mean that we should always use class properties instead of class methods? Has that been decided?

Reply:
My first question was how can you use `this` inside that function if it was not bound?

Regarding class properties instead of class methods, what is the outcome of this? https://github.com/elifesciences/elife-xpub/issues/201

Reply:
You only need to bind `this` if you're passing the function as an argument (or prop). If you're calling it as `this.whatever()` then that implicitly binds `this`. It's weird but that's how JS was designed.

Reply:
What if how we call `setData` changes? Is it safe to leave it like this? I would expect this to lead to bugs.

Reply:
> I would expect this to lead to bugs.

Yes, `this` is a footgun. Continuing the discussion over at https://github.com/elifesciences/elife-xpub/issues/201#issuecomment-397652378

Reply:
Ok, I guess this can go for now and we continue discussion there. I assume you tested and it works, everything else LGTM.